cwl/src/main/scala/cwl/CwlExpressionCommandPart.scala

```diff
@@ -43,3 +44,29 @@ case class CwlArgumentCommandPart(argument: CommandLineBinding) extends CommandP
   }
 }
 
+case class InputParameterCommandPart(argument: CommandInputParameter) extends CommandPart {
+
+  override def instantiate(inputsMap: Map[LocalName, WomValue],
+                           functions: IoFunctionSet,
+                           valueMapper: (WomValue) => WomValue) = {
+
+    val localInputsMap = inputsMap.map{
+      case (LocalName(localName), WomSingleFile(path)) => localName -> WomString(path)
+      case (LocalName(localName), value) => localName -> value
+    }
+
+
+    val womValue: WomValue = argument match {
+      case CommandInputParameter(id, _,_,_,_,_, Some(CommandLineBinding(None,Some(_),None,None,None,None,None)),_,_) =>
```

Comment:
I wonder if at some point it just becomes better to just do `if blah.isDefined`, `blah.get` :smile:

Reply:
This is done because I have nested options, i.e. command line binding and the position of the command line binding.

I can't match the case where there is a command line binding *and* the command line binding has a position binding with an if statement but I can with a pattern match.

Reply:
The Nones are probably too strict.  I'm looking to see what I can make `_`

Reply:
What about adding a method on `CommandInputParameter` like 
```scala
def hasPosition = inputBinding.exists(_.position.isDefined)
```

Reply:
So it turns out all those `None`'s being enforced is correct.  This case only handles when the input is bound by the id and not by any of the (many) options in `CommandLineBinding`.

Now I could write a function enforcing all this too, but it would only ever be used here and I would use a pattern match to implement that, too.  

I promise I would make this cleaner if I could :).